client/my-sites/marketplace/controller.js

```diff
@@ -8,25 +8,34 @@ import page from 'page';
  * Internal dependencies
  */
 import { getSiteFragment } from 'calypso/lib/route';
-import MarketplacePluginDetails from 'calypso/my-sites/marketplace/marketplace-plugin-details';
-import MarketplaceDomainUpsell from 'calypso/my-sites/marketplace/marketplace-domain-upsell';
-import MarketplacePluginSetup from 'calypso/my-sites/marketplace/marketplace-plugin-setup-status';
-import MarketplaceStandaloneThankYou from 'calypso/my-sites/marketplace/marketplace-stand-alone-thank-you';
-import MarketplaceTest from 'calypso/my-sites/marketplace/marketplace-test';
+import MarketplacePluginDetails from 'calypso/my-sites/marketplace/pages/marketplace-product-details';
+import MarketplaceDomainUpsell from 'calypso/my-sites/marketplace/pages/marketplace-domain-upsell';
+import MarketplacePluginSetup from 'calypso/my-sites/marketplace/pages/marketplace-plugin-setup-status';
+import MarketplaceStandaloneThankYou from 'calypso/my-sites/marketplace/pages/marketplace-stand-alone-thank-you';
+import MarketplaceTest from 'calypso/my-sites/marketplace/pages/marketplace-test';
+import { getDefaultProductInProductGroup } from 'calypso/my-sites/marketplace/marketplace-product-definitions';
 
-export function renderMarketplacePlugin( context, next ) {
-	const siteUrl = getSiteFragment( context.path );
-
-	const pluginSlug = decodeURIComponent( context.params.plugin );
+export function renderMarketplaceProduct( context, next ) {
+	const siteFragment = getSiteFragment( context.path );
+	const { productSlug: productSlugParam, productGroupSlug: productGroupSlugParam } = context.params;
+	let productSlug = productSlugParam ? decodeURIComponent( productSlugParam ) : null;
+	const productGroupSlug = productGroupSlugParam
+		? decodeURIComponent( productGroupSlugParam )
+		: null;
 
+	if ( ! siteFragment ) {
+		return page.redirect( '/home' );
+	} else if ( ! productGroupSlug && ! productSlug ) {
+		return page.redirect( `/home/${ siteFragment }` );
```

Comment:
Is there a good reason why we use `page.redirect()` instead of just `page()` or even `navigate()` from `lib/navigate` that was recently introduced?

Reply:
Hmmm this is a mistake, and thanks for pointing me to navigate!